src/test/java/hudson/security/LDAPSecurityRealmTest.java

```diff
@@ -433,6 +433,53 @@ public void customBeanBindingHindersMultiServerConfig() throws IOException, SAXE
 
     }
 
+    @Test
+    public void configRoundTripEnvironmentProperties() throws Exception {
+        final String server = "ldap.itd.umich.edu";
+        final String rootDN = "ou=umich,ou.edu";
+        final String userSearchBase = "cn=users,ou=umich,ou.edu";
+        final String managerDN = "cn=admin,ou=umich,ou.edu";
+        final String managerSecret = "secret";
+
+        LDAPConfiguration c = new LDAPConfiguration(server, rootDN, false, managerDN, Secret.fromString(managerSecret));
+
+        LDAPSecurityRealm.EnvironmentProperty[] environmentProperties = {new LDAPSecurityRealm.EnvironmentProperty("java.naming.ldap.typesOnly", "true")};
+        c.setEnvironmentProperties(environmentProperties);
+        c.setUserSearchBase(userSearchBase);
+
+        List<LDAPConfiguration> configurations = new ArrayList<LDAPConfiguration>();
+        configurations.add(c);
+        LDAPSecurityRealm realm = new LDAPSecurityRealm(
+                configurations,
+                false,
+                null,
+                IdStrategy.CASE_INSENSITIVE,
+                IdStrategy.CASE_INSENSITIVE
+        );
+
+        r.jenkins.setSecurityRealm(realm);
+
+        final JenkinsRule.WebClient client = r.createWebClient();
+        r.submit(client.goTo("configureSecurity").getFormByName("config"));
+
+        LDAPSecurityRealm newRealm = (LDAPSecurityRealm) r.jenkins.getSecurityRealm();
+        assertNotSame(realm, newRealm);
+        LDAPConfiguration config = newRealm.getConfigurations().get(0);
```

Comment:
Maybe newConfig would make it clearer that is the configuration obtained from newRealm?